src/utils/atoms/combineAtoms.ts

```diff
@@ -0,0 +1,16 @@
+import { createAtom } from '@reatom/core';
+import type { Atom } from '@reatom/core';
+
+type AtomsMap = { [key: string]: Atom<any> };
+
+export function combineAtoms<T extends AtomsMap>(
+  shape: T,
+): Atom<{ [key in keyof T]: T[key] extends Atom<infer S> ? S : never }> {
+  return createAtom(shape, ({ get }, state = {}) => {
+    return Object.keys(shape).reduce((acc, key) => {
+      // @ts-ignore
+      acc[key] = get(key);
+      return acc;
+    }, {} as { [key in keyof T]: T[key] extends Atom<infer S> ? S : never });
```

Comment:
isn't it unknown instead of never here ?
if so, it's possible to go this way, to eliminate // @ts-ignore
```

export function combineAtoms<T extends AtomsMap>(
  shape: T,
): Atom<{ [key in keyof T]: T[key] extends Atom<infer S> ? S : unknown }> {
  return createAtom(shape, ({ get }, state = {}) => {
    return Object.keys(shape).reduce((acc, key) => {
      acc[key as keyof T] = get(key);
      return acc;
    }, {} as { [key in keyof T]: T[key] extends Atom<infer S> ? S : unknown });
  });
}

```

Reply:
Nice catch!